paimon-core/src/main/java/org/apache/paimon/mergetree/compact/FirstRowMergeTreeCompactRewriter.java

```diff
@@ -77,8 +80,19 @@ protected boolean rewriteChangelog(
     }
 
     @Override
-    protected boolean upgradeChangelog(int outputLevel, DataFileMeta file) {
-        return file.level() == 0;
+    protected UpgradeStrategy upgradeChangelog(int outputLevel, DataFileMeta file) {
```

Comment:
maybe this overwrite is no need

Reply:
this class is not extended lookup rewriter.